pkg/utils/bitutils.go

```diff
@@ -208,9 +208,21 @@ func UnsafeByteSliceToString(haystack []byte) string {
 	return *(*string)(unsafe.Pointer(&haystack))
 }
 
-func NormalizeSearchBytes(isCaseInSensitive bool, valueBytes []byte) []byte {
+func BytesCaseInsensitiveEqual(a, b []byte) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for i := range a {
+		if a[i] != b[i] && a[i] != b[i]^32 {
+			return false
+		}
+	}
```

Comment:
I don't think this case-insensitive equality is constrained enough. I'm pretty sure you'll get `[]byte{'^'} == []byte{'~'}` since `^` is 94 in ASCII and `~` is 126 (94 + 32)